docker/docker_image_src.go

```diff
@@ -197,7 +197,7 @@ func (s *dockerImageSource) fetchManifest(ctx context.Context, tagOrDigest strin
 	logrus.Debugf("Content-Type from manifest GET is %q", res.Header.Get("Content-Type"))
 	defer res.Body.Close()
 	if res.StatusCode != http.StatusOK {
-		return nil, "", errors.Wrapf(registryHTTPResponseToError(res), "Error reading manifest %s in %s", tagOrDigest, s.physicalRef.ref.Name())
```

Comment:
There’s a history and experience to this, see https://github.com/containers/image/pull/420 .

Even without that and focusing _only_ on the `copy` caller, this loses data:
- about which mirror is used
- about which manifest (for a multi-arch image) is failing

It might make sense to make this more intelligent, and add that context only in the cases where that data is not the defaults — but consider the `skopeo inspect` example from #420. This might need a clearer definition about which context is expected to be added by the caller vs. callee.

Reply:
Well we could try to do this at the higher level, where the other stutter happens, but I was not sure if this would remove data for other transports.

Reply:
That’s why I ended up musing about caller/callee expectations.

Both c/image/copy and c/image/docker added all of that context because each of those sub packages found all of that context helpful. I mean, we _could_ define a `c/image/internal/errors.ManifestReadingError{manifestReference reference.Named}` and have this case return a typed error that the copy code could recognize and process specially (it would be a special case, but a clean and IDE-visible special case) — but at that point thinking about the caller/callee context expectations almost seems easier.

Reply:
Reverted this back, and will concentrate on removing "error" and "Error" from obvious error messages.